data-prepper-plugins/s3-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/configuration/ObjectKeyOptions.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.sink.configuration;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * An implementation class of path prefix and file pattern configuration Options
+ */
+public class ObjectKeyOptions {
+    private static final String DEFAULT_OBJECT_NAME_PATTERN = "events-%{yyyy-MM-dd'T'hh-mm-ss}";

```

Comment:
This value should be:

```
events-%{yyyy-MM-dd'T'hh-mm-ss}-${epochSeconds}-${uniqueId}.${extension}
```

I think there may be some broad issues here about format strings as well.

So, my suggestion is that what we do for now is remove the `name_pattern` option from customers. We can consider making this configurable later. For this PR, please always create the pattern as the following.

```
events-%{yyyy-MM-dd'T'hh-mm-ss}-${epochSeconds}-${uniqueId}.${extension}
```

The `epochSeconds` will the the same time as the string in `yyyy-MM-dd'T'hh-mm-ss`, but represented as seconds since the 1970 epoch. The unique Id is generated once per Data Prepper instance. And the `extension` comes from the codec.

Reply:
Resolved.